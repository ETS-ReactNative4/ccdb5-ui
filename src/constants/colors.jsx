/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
