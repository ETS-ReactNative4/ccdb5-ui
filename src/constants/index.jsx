// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
