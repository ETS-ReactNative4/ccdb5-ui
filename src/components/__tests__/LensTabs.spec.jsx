import configureMockStore from 'redux-mock-store'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import {
  LensTabs,
  mapDispatchToProps,
  mapStateToProps
} from '../Trends/LensTabs'
import React from 'react'
import renderer from 'react-test-renderer'
import { REQUERY_ALWAYS } from '../../constants'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme'

function setupSnapshot() {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {} )

  return renderer.create(
    <Provider store={ store }>
      <IntlProvider locale="en">
        <LensTabs lens={ 'Product' } subLens={ 'Product' }/>
      </IntlProvider>
    </Provider>
  )
}

describe( 'component:LensTabs', () => {
  it( 'renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  describe( 'buttons', () => {
    let cb = null
    let target = null

    beforeEach( () => {
      cb = jest.fn()
      target = shallow( <LensTabs onTab={ cb }
                                  lens={ 'Product' }
                                  subLens={ 'Issue' }/> )
    } )

    it( 'tabChanged is called with Product when the button is clicked', () => {
      const prev = target.find( '.tabbed-navigation button.sub_product' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalledWith( 'sub_Product' )
    } )

    it( 'tabChanged is called with Issue when the button is clicked', () => {
      const prev = target.find( '.tabbed-navigation button.issue' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalledWith( 'Issue' )
    } )
  } )

  describe( 'mapDispatchToProps', () => {
    it( 'hooks into changeDataSubLens', () => {
      const dispatch = jest.fn()
      mapDispatchToProps( dispatch ).onTab( 'What' )
      expect( dispatch.mock.calls ).toEqual( [
        [ {
          requery: REQUERY_ALWAYS,
          subLens: 'what',
          type: 'DATA_SUBLENS_CHANGED'
        } ]
      ] )
    } )
  } )

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        query: {
          lens: 'foo',
          subLens: 'bar'
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( { lens: 'foo', subLens: 'bar' } )
    } )
  } )

} )
