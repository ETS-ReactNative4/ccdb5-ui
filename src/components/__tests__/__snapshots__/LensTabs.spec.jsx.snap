// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:LensTabs renders Product without crashing 1`] = `
<div
  className="tabbed-navigation lens"
>
  <h2>
    Product trends for selected criteria
  </h2>
  <section>
    <button
      className="tab sub_product active"
      onClick={[Function]}
    >
      Sub-products
    </button>
    <button
      className="tab issue"
      onClick={[Function]}
    >
      Issues
    </button>
  </section>
</div>
`;
